src/message.ts

```diff
@@ -0,0 +1,79 @@
+import { Promise } from 'es6-promise';
+import { TALKER_TYPE } from './strings';
+import Talker from './index';
+
+abstract class Message {
+  protected readonly type: string = TALKER_TYPE;
+
+  constructor(
+    /*
+     * @property talker - A {@link Talker} instance that will be used to send responses
+     */
+    protected readonly talker: Talker,
+    /*
+     * @property namespace - A namespace to with which to categorize messages
+     */
+    public readonly namespace: string,
+    public readonly data: Stringifyable,
+    public readonly responseToId: number | null = null
+  ) {}
+}
+
+export interface JSONifiedMessage {
```

Comment:
JSON is a string serialization format; if this is still an object, then it's not JSONified...

```suggestion
export interface JSONableMessage {
```

or

```suggestion
export interface SerializableMessage {
```